src/esm_master/task.py

```diff
@@ -534,3 +550,181 @@ def output_steps(self):
+                    repo["is_repo_operation"] = True
+                    self.executed_repo_commands.append(command)
+                    if self.num_get_commands == len(self.executed_repo_commands):
+                        repo["is_last_repo"] = True
+                    break
+
+        return repo
+
+    def add_repo_error(self, command, repo, error):
+        """
+        If an error occurred during the repo ``command``, then stores information about
+        the error, and the associated package. The information about the error is
+        saved in the attribute ``self.repo_errors`` (if it doesn't exist, it creates
+        it).
+
+        The ``repo_errors`` attribute itself is a ``dict`` with keys being the failed
+        ``command``s, each of them containing the following variables:
+        - ``model``: model name
+        - ``repo``: url of the repo
+        - ``destination``: destination folder
+        - ``error``: "destination path exist" if the destination exist, and the value
+          of the ``error`` parameter if it the destination does not exist
+
+        Note: because the ``error`` is caught from an ``except`` of a ``subprocess``
+        command, the actual error that occurred during the execution of the ``command``
+        by ``subprocess.run`` is not caught. That means that we cannot evaluate which
+        type of error occurred during the repo command (i.e. was an error related to
+        access to the repo, or was it the error that happens when the folder already
+        exist?). To compensate for this problem, this method also checks whether the
+        destination folder exist or not to save a ``"destination path exist"`` error
+        description or to keep the same error string as reported by ``error``
+        (permission problem). This approach is not ideal. It would be possible to catch
+        the ``subprocess`` error, by capturing the stdout/stderr of the ``subprocess``
+        command, but then, the downloading progress is also captured and not displayed
+        to the user, which might make the users wonder if ESM-Tools is stalling.
+
+        Parameters
+        ----------
+        command : str
+            Repo command for which the error might have occur
+        repo : dict
+            Information associated to the repo ``command``, including tha ``package``
+            object
+        error : str
+            Error caught from ``except``
+        """
+        # Initializes the list of executed repo commands
+        if not hasattr(self, "repo_errors"):
+            self.repo_errors = {}
+
+        destination = repo["package"].destination
+        # Checks whether the destination path exists to save a "destination path exist"
+        # error
+        if os.path.isdir(destination):
+            error = "destination path exists"
+        for folder in self.folders_after_download:
+            if destination in folder:
+                full_destination = folder
```

Comment:
I don't entirely understand this part...?

Reply:
Correction: What happens is clear, but I do not understand why it needs to happen....

Reply:
the user tries to install awicm3 but has not permissions for oifs, xios and rnfmap, however, fesom clones because it has no restirctions. The user then gets the persmission error for oifs, xios, rnfmap, contacts Jan or myself and we grant the necessary permisions to the user. Then the user runs the command again and can clone oifs, xios and rnfmap, but git returns an error for fesom because the destination already exist. With this, we can catch this problem and tell the user what to do.